src/api/db/data/20250131094818_nullify_targets_on_bs_request_actions.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class NullifyTargetsOnBsRequestActions < ActiveRecord::Migration[7.0]
+  def up
+    bs_request_actions = BsRequestAction.where('target_project_id IS NOT NULL AND target_package_id IS NULL')
+    bs_request_actions.in_batches do |batch|
+      batch.find_each do |action|
+        target_project = Project.find_by_name(action.target_project)
+        if target_project.nil?
+          action.update(target_project_id: nil)
```

Comment:
This will silently fail and run a lot of validations etc. Better to use `update_attribute!`

Reply:
`update_attribute!` is not available in Rails 7.0...

Reply:
and that will still run a lot of validations etc. 

https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update-21

Reply:
> and that will still run a lot of validations etc.
> 
> https://api.rubyonrails.org/v7.0.3/classes/ActiveRecord/Persistence.html#method-i-update_attribute

Sure. However:
- The number of records which will be updated is relatively low. Most of the time of the data migration will be taken to search for removed projects or packages.
- There are not so many validations: https://github.com/openSUSE/open-build-service/blob/434c4b5333bbd09063474f1530d478884e95a3ae/src/api/app/models/bs_request_action.rb#L39-L41

Reply:
Potentially

```
[1] pry(main)> BsRequestAction.where('target_project_id IS NOT NULL AND target_package_id IS NULL').count
=> 257361
[2] pry(main)> BsRequestAction.where('target_project_id IS NOT NULL AND target_package_id IS NOT NULL').count
=> 1031538
```

and it's not only validations, it's callbacks, updated_at columns etc. etc.

Reply:
Changed to `update_attribute`.